lib/fabrik/DirectorInstance.js

```diff
@@ -50,37 +67,62 @@ class DirectorInstance extends BaseInstance {
         this.networkSegmentIndex = networkSegmentIndex;
       })
       .tap(() => {
+        //TODO: This still has coupling with space guid as it calls ensureTenantId. 
+        //As part of SF 2.0, we can go to ETCD and fetch the tenant_id which will 
+        //ensure we dont call CF again. For now, we add a check for the feature validation.
         if (operation.type === 'delete') {
           return Promise
             .all([
-              this.deleteSecurityGroup(),
-              this.ensureSpaceGuid()
-              .then(space_guid => this.manager.deleteRestoreFile(space_guid, this.guid))
-              .catch(err => {
-                logger.error(`Failed to delete restore file of instance '${this.guid}'`);
-                logger.error(err);
-                throw err;
-              })
+              this.platformManager.preInstanceDeleteOperations({
+                ruleId: this.securityGroupName
+              }),
+              this.deleteRestoreFile()
               //ScheduledBackupJob is not being cancelled on delete, rather BackupJob itself takes care of it
             ]);
         }
       });
   }
 
+  deleteRestoreFile() {
+    Promise.try(() => {
+      if (_.includes(this.manager.agent.features, 'backup')) {
+        return this.platformManager.ensureTenantId()
+          .then(space_guid => this.manager.deleteRestoreFile(space_guid, this.guid))
+          .catch(err => {
+            logger.error(`Failed to delete restore file of instance '${this.guid}'`);
+            logger.error(err);
+            throw err;
+          });
+      }
+    });
+  }
+
   finalize(operation) {
     const action = _.capitalize(operation.type);
     return Promise
       .try(() => {
-        switch (operation.type) {
-        case 'create':
-          return this
-            .createSecurityGroup(operation.space_guid)
-            .tap(() => operation.state === CONST.OPERATION.SUCCEEDED ? this.scheduleAutoUpdate() : {});
-        case 'update':
-          return this.ensureSecurityGroupExists();
+        if (operation.context.platform === CONST.CONSUMING_ENVIRONMENT.CF) {
```

Comment:
The intention of having platformmanager was to avoid such checks within Instance managers. If such checks cannot be avoided then better not to have platform manager, as makes code a bit difficult to read thru, because one would think and ask what would happen to K8S as part of post creation as the code omits it out.

Reply:
sorry. forgot to remove this. This is not at all required :P